owncloudApp/src/main/java/com/owncloud/android/presentation/files/filelist/MainFileListFragment.kt

```diff
@@ -556,7 +561,13 @@ class MainFileListFragment : Fragment(),
                     val appProviderItemView = BottomSheetFragmentItemView(requireContext())
                     appProviderItemView.apply {
                         title = getString(R.string.ic_action_open_with_web, appRegistryProvider.name)
-                        itemIcon = ResourcesCompat.getDrawable(resources, R.drawable.ic_open_in_web, null)
+                        itemIcon = try {
+                            removeDefaultTint()
+                            getDrawableFromUrl(context, appRegistryProvider.icon)
+                        } catch (e: Exception) {
+                            addDefaultTint(R.color.bottom_sheet_fragment_item_color)
+                            ResourcesCompat.getDrawable(resources, R.drawable.ic_open_in_web, null)
+                        }
```

Comment:
What kind of exceptions can Glide throw and in which cases?

Reply:
Glide could catch an exception when the url is not correct or is an empty String. I don't know if we could actually receive a wrong url from the server, but it was to cover that possibility.
There is another simpler option: add the error function to the Glide build and put the drawable default icon directly. But we must also add the `addDefaultTint()`. So we would have to create a function adding the default icon and the `addDefaultTint()` function. The problem is that this last function is only found in the context of `BottomSheetFragmentItemView`, therefore a parameter would have to be added to the `getDrawableFromUrl `function, which would result in this function depending on `BottomSheetFragmentItemView` to work, which would not be correct. I would leave it with the try catch.

Reply:
At least add a log to see which is the exception in case it happens